File: backend/src/main/java/com/delivery/bid/entity/Bid.java
package com.delivery.bid.entity;

import com.delivery.store.entity.Store;
import com.delivery.auction.entity.Auction;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "bids")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Bid {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Double amount;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BidStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", nullable = false)
    private Store store;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_id", nullable = false)
    private Auction auction;
}

File: frontend/src/interfaces/Bid.ts
import {Store} from "./Store"
import {Auction} from "./Auction"

export enum BidStatus {
    WON = 'WON',
    OUTBID = 'OUTBID',
}

export interface Bid {
    id: number;
    amount: number;
    timestamp: string;
    status: BidStatus;
    store: Store;
    auction: Auction;
}